akka-actor/src/main/scala/akka/io/Dns.scala

```diff
@@ -87,29 +87,35 @@ object Dns extends ExtensionId[DnsExt] with ExtensionIdProvider {
   override def get(system: ActorSystem): DnsExt = super.get(system)
 }
 
-class DnsExt(val system: ExtendedActorSystem) extends IO.Extension {
-
-  val Settings = new Settings(system.settings.config.getConfig("akka.io.dns"))
-
-  class Settings private[DnsExt] (_config: Config) {
+/**
+ * Load the resolver from akka.io.dns.<resolverName> with the given <managerName>
+ */
+class DnsExt(val system: ExtendedActorSystem, resolverName: String, managerName: String) extends IO.Extension {
```

Comment:
Hmm, this is an extension, which the hole point of is to have an actor-system singleton, so isn't it weird that it will be possible to create multiple instances? Should that rather be a separate thing, and then this one inherits it? Am I missing something clever here?

Reply:
Yeah this feels wrong.

The one tricky thing is that the pluggable DNS resolver creates a user class (the manger) and passes its self into the constructor.

However thinking about it some more we don't want to allow arbitrary DnsExt to be created, only a `async-dns` resolver one for management. So maybe I should open it up more

Reply:
But that is also quite ugly as we don't want to support users creating an async dns resolver manually. 

Maybe service discovery + zero dep impls should live in Akka.